pkg/sync/syncWorker.go

```diff
@@ -342,7 +344,14 @@ func (s *syncWorker) setupStream(
 			err,
 		)
 	}
-	originatorStream := &originatorStream{nodeID: nodeID, stream: stream}
+	expBackoff := backoff.NewExponentialBackOff()
```

Comment:
setupStream itself is wrapped in a backoff. Which itself is wrapped in an endless loop. Why do we need a third layer of retries?

Reply:
We should be able to retry saving an individual message without needing to go back to the other node and retrieve a fresh page of results.